client/src/main/java/io/pravega/client/ClientFactory.java

```diff
@@ -60,12 +60,22 @@
      *
      * @param scope The scope string.
      * @param controllerUri The URI for controller.
+     * @param creds Optional credentials to connect to controller.
      * @return Instance of ClientFactory implementation.
      */
-    static ClientFactory withScope(String scope, URI controllerUri) {
-        val connectionFactory = new ConnectionFactoryImpl(false);
+    static ClientFactory withScope(String scope, URI controllerUri, PravegaCredentials creds) {
+
+        val connectionFactory = new ConnectionFactoryImpl();
+        boolean enableTls = Boolean.parseBoolean(System.getProperty("io.pravega.auth.enabled"));
+        String tlsCertFile = System.getProperty("io.pravega.auth.certfile");
+
         return new ClientFactoryImpl(scope, new ControllerImpl(controllerUri, ControllerImplConfig.builder().build(),
-                connectionFactory.getInternalExecutor()), connectionFactory);
+                connectionFactory.getInternalExecutor(), creds, enableTls, tlsCertFile), connectionFactory);
```

Comment:
I think the comment about `enableTls` makes sense. We can retrieve it from the URL.  Will update the code to do so. Have already dropped it from the parameters and was retrieving it from system properties as a result of another comment. `tlsCertFile` is an choice for user to override the default truststore. If this parameter is null - we use the default trust store. The latest code even this is not present as a parameter.

Reply:
Can we infer the SSL aspect on the GRPC connections too?  I believe so but wanted to check.  In other words, if the controller url is   `tcp://...` then no ssl, but if `tls://` then yes.. ?   This way this works for both the RestServer and the grpc stuff?

Reply:
I think it would be fine to use `http:` and `https:` as the scheme for both types of connections, because GRPC is actually based on HTTP/2 which still uses that scheme.  WDYT?